src/PackagePrivate/MediaWikiNormalizedTermCleaner.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+namespace Wikibase\TermStore\MediaWiki\PackagePrivate;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Wikimedia\Rdbms\DBError;
+use Wikimedia\Rdbms\IDatabase;
+use Wikimedia\Rdbms\ILoadBalancer;
+
+/**
+ * Cleans up the normalized term store after some terms are no longer needed.
+ * Unused term_in_lang, text_in_lang and text rows are automatically removed.
+ * (Unused type rows are never cleaned up.)
+ *
+ * @license GPL-2.0-or-later
+ */
+class MediaWikiNormalizedTermCleaner {
+
+	/** @var ILoadBalancer */
+	private $lb;
+
+	/** @var IDatabase a connection to DB_REPLICA */
+	private $dbr = null;
+
+	/** @var IDatabase a connection to DB_MASTER */
+	private $dbw = null;
```

Comment:
I'd go for masterDatabase an replicateDatabase or something similar. The current abbreviations are intuitive to you and me because we have worked with MW a lot already.